File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/Cat.java
package io.quarkus.it.panache;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

@Entity
public class Cat extends PanacheEntity {

    String name;

    @ManyToOne
    CatOwner owner;

    Double weight;

    public Cat(String name, CatOwner owner) {
        this.name = name;
        this.owner = owner;
    }

    public Cat() {
    }
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/Cat.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntity
import jakarta.persistence.Entity
import jakarta.persistence.ManyToOne

@Entity
class Cat @JvmOverloads constructor(
    var name: String? = null,
    @ManyToOne
    var owner: CatOwner? = null,
    var weight: Double? = null
) : PanacheEntity() {
    companion object : PanacheCompanion<Cat>

    constructor(owner: CatOwner?) : this(null, owner, null)
}
